### PR TITLE
Add seed-demo tool for rich placeholder content

### PR DESCRIPTION
**Pull request by jonhickman**
## Summary

Adds a new `seed-demo` command (single-command CLI + interactive mode) that seeds a Ghost site with rich, realistic demo content in a single run. It builds on the ideas in `random-posts` but produces a much fuller, theme-testable site.

In one run it can create:
- **Posts** with feature images and weighted in-post image cards (0-3 per post, never before the first paragraph)
- A **distributed tag set** (5 fixed tags + up to 30 extra) plus a "Collection" tag archive
- An **`/about` page**
- A **"golden" Style Guide post** that exercises every editor card, a full inline-formatting/typography sampler, and a set of edge cases, so it doubles as a front-end theme stress test. Card shapes mirror `@tryghost/kg-default-nodes` so everything stays editable in Admin.
- A **navigation menu** (Home, About, Style Guide, Author, Collection)
- Optional **date distribution** across a range (spread-with-jitter), so posts look naturally paced
- An optional **dummy author**, attributed as primary on a configurable share of posts and as second author on the Style Guide

## Auth model

- Standard content operations work with a normal integration Admin API key.
- Writing **navigation** (a settings write) and creating the **dummy author** (a staff user, created via Universal Import so no invite email is sent) require an **owner/admin staff access token** — same `{id}:{secret}` format.
- With an integration key, those two steps **skip gracefully** and print paste-able navigation JSON / importable author JSON instead of failing the run. This mirrors how other staff-scoped tools behave.

## Notes

- Assets are sourced from permissively licensed providers (Lorem Picsum for images, lorem.media for audio/video) and **re-hosted on the target site** so the result is self-contained. Licensing is documented in the README.
- API calls are rate-limited following the existing `random-posts` pattern (`Promise.mapSeries` + `Promise.delay`, `concurrent: 1`).
- The author step is idempotent (reuses an existing author by slug/email); `/about`, the Style Guide, and navigation are matched by slug and updated rather than duplicated. Posts are always added.

## Test plan

- [x] `yarn test` (119 passing, including new unit tests for the Lexical card builders, tag pool, date distribution, author module, and Style Guide assembly)
- [x] `yarn lint` clean
- [x] Live round-trip against a local Ghost instance:
  - [x] Full run: 100 posts + 10 extra tags + author + about + style guide + navigation
  - [x] Dummy author created with a hosted profile image; primary on the expected share of posts; second author on the Style Guide
  - [x] Idempotent re-run reuses the existing author (no duplicate)
  - [x] Integration key: navigation + author steps skip gracefully and print paste-able/importable JSON
  - [x] Interactive mode walked end-to-end

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: lib/demo/assets.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import {join} from 'node:path';
+import {tmpdir} from 'node:os';
+import {mkdtemp, writeFile, rm} from 'node:fs/promises';
+import axios from 'axios';
+import Promise from 'bluebird';
```
[the remark this answers is not in the document]
Comment (PaulAdamDavis) [reply lib/demo/assets.js]: I agree, and this applies for all changes in this PR. We've been moving away from Bluebird for a while, as native promises can do everything we need.

File: lib/demo/assets.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
+import {join} from 'node:path';
+import {tmpdir} from 'node:os';
+import {mkdtemp, writeFile, rm} from 'node:fs/promises';
+import axios from 'axios';
```
Comment (PaulAdamDavis) [inline lib/demo/assets.js]: Node 22+ has `fetch` natively, which we should use here. It should be able to do everything `axios` is doing, but with one less dependency.

This applies across the whole PR.

File: lib/demo/author.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+// Error code signalling that the dummy author could not be created because the
+// supplied key is an integration token. Creating a staff user requires writing
+// to /db/ (Universal Import), which only a staff access token can do — mirrors
+// NAV_NO_PERMISSION in navigation.js.
+const AUTHOR_NO_PERMISSION = 'AUTHOR_NO_PERMISSION';
+
+const baseUrl = options => options.apiURL.replace(/\/$/, '');
+const dbUrl = options => `${baseUrl(options)}/ghost/api/admin/db/`;
+
+const slugify = name => (name || '')
```
Comment (PaulAdamDavis) [inline lib/demo/author.js]: `@tryghost/string` has a built-in `slugify` method that we should use to be consistent with other tools.

File: tasks/seed-demo.js [hunk 4]
```diff
@@ -0,0 +1,325 @@
+import Promise from 'bluebird';
```
[the remark this answers is not in the document]
Comment (PaulAdamDavis) [reply tasks/seed-demo.js]: Use native promises here

**Review by PaulAdamDavis — CHANGES_REQUESTED** [review]
The main issue for me is using Bluebird; we should use native promises instead.
